packages/xrpl/src/models/transactions/vaultDeposit.ts

```diff
@@ -24,7 +24,8 @@ export interface VaultDeposit extends BaseTransaction {
   /**
    * Asset amount to deposit.
    */
-  Amount: Amount
+  // TODO: remove MPTAmount when MPTv2 is released
+  Amount: Amount | MPTAmount
```

Comment:
Its confusing to keep track of what `Amount` really means -- this could change with time.

I'd prefer to spell out all the constituent types, instead of using the `Amount` alias. This improves readability vastly and reduces one more indirection.
```suggestion
  Amount: MPTAmount | IssuedCurrencyAmount | string
```

Reply:
That's fair but these other transactions will support MPT its v2 is released. We can just add it as a union to `Amount` when that happens.